src/shogun/multiclass/tree/RandomCARTree.cpp

```diff
@@ -49,238 +49,22 @@ void CRandomCARTree::set_feature_subset_size(int32_t size)
 	m_randsubset_size=size;
 }
 
-int32_t CRandomCARTree::compute_best_attribute(SGMatrix<float64_t> mat, SGVector<float64_t> weights, SGVector<float64_t> labels_vec,
-	SGVector<float64_t> left, SGVector<float64_t> right, SGVector<bool> is_left_final, int32_t &num_missing_final, int32_t &count_left,
-														 int32_t &count_right)
-{
-	int32_t num_vecs=mat.num_cols;
-	int32_t num_feats=mat.num_rows;
-
-	int32_t n_ulabels;
-	SGVector<float64_t> ulabels=get_unique_labels(labels_vec,n_ulabels);
-
-	// if all labels same early stop
-	if (n_ulabels==1)
-		return -1;
-
-	SGVector<float64_t> total_wclasses(n_ulabels);
-	total_wclasses.zero();
+int32_t CRandomCARTree::compute_best_attribute(const SGMatrix<float64_t>& mat, const SGVector<float64_t>& weights, const SGVector<float64_t>& labels_vec,
+	SGVector<float64_t>& left, SGVector<float64_t>& right, SGVector<bool>& is_left_final, int32_t &num_missing_final, int32_t &count_left,
+	int32_t &count_right, int32_t subset_size)
 
-	SGVector<int32_t> simple_labels(num_vecs);
-	float64_t delta=0;
-	if (m_mode==PT_REGRESSION)
-		delta=m_label_epsilon;
-
-	for (int32_t i=0;i<num_vecs;i++)
-	{
-		for (int32_t j=0;j<n_ulabels;j++)
-		{
-			if (CMath::abs(labels_vec[i]-ulabels[j])<=delta)
-			{
-				simple_labels[i]=j;
-				total_wclasses[j]+=weights[i];
-				break;
-			}
-
-		}
-	}
-
-	REQUIRE(m_randsubset_size<=num_feats, "The Feature subset size(set %d) should be less than"
-	" or equal to the total number of features(%d here)\n",m_randsubset_size,num_feats)
+{
+	REQUIRE(m_randsubset_size<=mat.num_rows, "The Feature subset size(set %d) should be less than"
+	" or equal to the total number of features(%d here)\n",m_randsubset_size,mat.num_rows)
```

Comment:
remove "here"
Also, please make really clear what things are here. Are you talking about subsets of data or of components?
Thirdly, is the user able to cause this error message?